### DISCUSSION TITLE
Ability to run a mixed test code base using node:test

### DISCUSSION THREAD

**Discussion by Raynos** [avajs/ava#3272] [Q&A]
I want to migrate my tests and write new tests using `node:test` and continue using `ava` to run the entire test file.

Is it possible for ava cli to detect that a given file runs `node:test` and hook into that, or hook into the TAP output ?

**Comment by Raynos** [comment]
I briefly looked at a jscodeshift to go from `ava` to `node:test` but that's non-trivial, i was looking at the `tape` to `ava` codemod if i could inverse it.

**Comment by novemberborn** [comment]
The oldest AVA tests actually use the tap runner. We run separate sets of tests.

It would be possible within AVAs config to filter test files yourself, so you wouldn't provide globs but resolve the list of files fit for AVA.

**Comment by Raynos** [comment]
Running seperate tests is possible I guess but a bit of a pita, would be nice if it worked as is with a single test runner,

I guess I can use `*.ava.ts` & `*.test.ts` or something for two different glob patterns
